src/sql/workbench/contrib/dashboard/browser/widgets/explorer/explorerWidget.component.ts

```diff
@@ -106,32 +107,54 @@ export class ExplorerWidget extends DashboardWidget implements IDashboardWidget,
 				}
 			)));
 		} else {
-			this._register(subscriptionToDisposable(this._bootstrap.metadataService.databases.subscribe(
-				data => {
-					// Handle the case where there is no metadata service
-					data = data || [];
-					if (isStringArray(data)) {
-						data = data.map(item => {
-							const dbInfo: DatabaseInfo = { options: {} };
-							dbInfo.options[NameProperty] = item;
-							return dbInfo;
-						});
+			// TODO: remove this ADS side workaround for SQL On-Demand and handle it in SQL Tools Service
+			// engineEditionId === 11 means this is a SQL on-demand server
+			if (this._bootstrap.connectionManagementService.connectionInfo.serverInfo.engineEditionId === 11) {
```

Comment:
Just use the enum : https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/api/common/sqlExtHostTypes.ts#L388